app/src/main/java/io/appium/settings/LocationTracker.java

```diff
@@ -200,23 +321,24 @@ private void startLocationUpdatesWithoutPlayServices() {
                 mLocationManager.requestLocationUpdates(
                         LocationManager.GPS_PROVIDER,
                         LOCATION_UPDATES_INTERVAL,
-                        MIN_DISTANCE_CHANGE_FOR_UPDATES, this);
-                mLocationProvider = LocationManager.GPS_PROVIDER;
+                        MIN_DISTANCE_CHANGE_FOR_UPDATES, gpsLocationListener);
+                mGpsLocationProviderName = LocationManager.GPS_PROVIDER;
                 Log.d(TAG, "GPS location provider is enabled. Getting FINE location");
-                return;
             } catch (SecurityException e) {
                 Log.e(TAG, "Appium Settings has no access to FINE location permission", e);
             }
         }
-        try {
-            mLocationManager.requestLocationUpdates(
-                    LocationManager.NETWORK_PROVIDER,
-                    LOCATION_UPDATES_INTERVAL,
-                    MIN_DISTANCE_CHANGE_FOR_UPDATES, this);
-            mLocationProvider = LocationManager.NETWORK_PROVIDER;
-            Log.d(TAG, "NETWORK location provider is enabled. Getting COARSE location");
-        } catch (SecurityException e) {
-            Log.e(TAG, "Appium Settings has no access to COARSE location permission", e);
+        if (isNetworkEnabled) {
```

Comment:
should it be `else if` or we may try to request both providers at the same time?